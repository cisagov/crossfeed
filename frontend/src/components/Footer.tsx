import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(4),
    width: '100%',
    color: 'white',
    textAlign: 'center',
    display: 'flex',
    flexFlow: 'row nowrap',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.palette.primary.main
  }
}));

export const Footer: React.FC = () => {
  const classes = useStyles();

  return (
    <footer className={classes.root}>
      Cybersecurity and Infrastructure Security Agency
    </footer>
  );
};
