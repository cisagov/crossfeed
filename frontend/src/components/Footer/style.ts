import { makeStyles } from '@mui/material';

export const useFooterStyles = makeStyles((theme) => ({
  footerBox: {
    position: 'absolute',
    bottom: 0,
    width: '100%',
    backgroundColor: theme.palette.primary.main
  },
  footerContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: '0 auto',
    maxWidth: '1444px'
  },
  footerLogo: {
    display: 'flex',
    justifyContent: 'center',
    height: '100%'
  },
  footerNavItem: {
    display: 'flex',
    justifyContent: 'center',
    padding: '0.5rem',
    borderLeft: '1px solid white',
    gridColumn: 'span 2'
  },
  footerNavLink: {
    color: 'white',
    alwaysUnderline: 'true',
    alignItems: 'center'
  }
}));
