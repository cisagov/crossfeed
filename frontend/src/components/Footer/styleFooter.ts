import { styled } from '@mui/material/styles';

const PREFIX = 'Footer';

export const footerClasses = {
  footerBox: `${PREFIX}-footerBox`,
  footerContainer: `${PREFIX}-footerContainer`,
  footerLogo: `${PREFIX}-footerLogo`,
  footerNavItem: `${PREFIX}-footerNavItem`,
  footerNavLink: `${PREFIX}-footerNavLink`
}
export const FooterRoot = styled('div')(({ theme }) => ({
  [`& .${footerClasses.footerBox}`]: {
    position: 'absolute',
    bottom: 0,
    width: '100%',
    backgroundColor: theme.palette.primary.main
  },
  [`& .${footerClasses.footerContainer}`]: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    margin: '0 auto',
    maxWidth: '1444px'
  },
  [`& .${footerClasses.footerLogo}`]: {
    display: 'flex',
    justifyContent: 'center',
    height: '100%'
  },
  [`& .${footerClasses.footerNavItem}`]: {
    display: 'flex',
    justifyContent: 'center',
    padding: '0.5rem',
    borderLeft: '1px solid white',
    gridColumn: 'span 2'
  },
  [`& .${footerClasses.footerNavLink}`]: {
    color: 'white',
    alwaysUnderline: 'true',
    alignItems: 'center'
  }
}));
