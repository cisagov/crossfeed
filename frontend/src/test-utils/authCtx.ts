import { AuthContextType } from 'context';
import { testUser } from './user';

export const authCtx: AuthContextType = {
  user: testUser,
  token: 'some-test-token-never-verified',
  currentOrganization: null,
  loading: false,
  login: jest.fn(),
  logout: jest.fn(),
  setUser: jest.fn(),
  setOrganization: jest.fn(),
  refreshUser: jest.fn(),
  setLoading: jest.fn(),
  apiGet: jest.fn(),
  apiDelete: jest.fn(),
  apiPatch: jest.fn(),
  apiPost: jest.fn(),
  apiPut: jest.fn(),
  maximumRole: 'user',
  touVersion: 'v1-user',
  userMustSign: false
};
