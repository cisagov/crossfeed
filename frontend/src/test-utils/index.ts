export * from './test-utils';
export * from './user';
export * from './organization';
