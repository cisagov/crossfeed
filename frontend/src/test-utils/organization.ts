import { Organization } from 'types';

export const testOrganization = {
  id: 'd7d6e913-0370-4f43-aebc-6bd727adc70e',
  name: 'test org',
  rootDomains: ['example.com'],
  ipBlocks: [],
  userRoles: [],
  scanTasks: [],
  isPassive: false,
  inviteOnly: false,
  granularScans: []
};
