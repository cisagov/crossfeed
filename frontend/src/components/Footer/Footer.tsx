import React from 'react';
import { Box, Grid, Link } from '@mui/material';
import { useAuthContext } from 'context';
import logo from '../../assets/crossfeed.svg';
import { useFooterStyles } from './style';

export const CrossfeedFooter: React.FC = (props) => {
  const { logout } = useAuthContext();
  const footerClasses = useFooterStyles(props);
  return (
    <Box className={footerClasses.footerBox}>
      <Grid className={footerClasses.footerContainer} container>
        <Grid className={footerClasses.footerLogo} item xs={12} sm={2}>
          <Link href="/">
            <img src={logo} alt="Crossfeed Icon Navigate Home" />
          </Link>
        </Grid>
        <Grid className={footerClasses.footerNavItem} item xs={12} sm={2}>
          <p>
            <Link className={footerClasses.footerNavLink} href="/">
              Home
            </Link>
          </p>
        </Grid>
        <Grid className={footerClasses.footerNavItem} item xs={12} sm={2}>
          <p>
            <Link
              className={footerClasses.footerNavLink}
              href="https://docs.crossfeed.cyber.dhs.gov/"
            >
              Documentation
            </Link>
          </p>
        </Grid>
        <Grid className={footerClasses.footerNavItem} item xs={12} sm={2}>
          <p>
            <Link
              className={footerClasses.footerNavLink}
              href="https://www.cisa.gov"
            >
              CISA Homepage
            </Link>
          </p>
        </Grid>
        <Grid className={footerClasses.footerNavItem} item xs={12} sm={2}>
          <p>
            <Link
              className={footerClasses.footerNavLink}
              href="mailto:vulnerability@cisa.dhs.gov"
            >
              Contact Us
            </Link>
          </p>
        </Grid>
        <Grid className={footerClasses.footerNavItem} item xs={12} sm={2}>
          <p>
            <Link
              className={footerClasses.footerNavLink}
              href="/"
              onClick={logout}
            >
              Logout
            </Link>
          </p>
        </Grid>
      </Grid>
    </Box>
  );
};
