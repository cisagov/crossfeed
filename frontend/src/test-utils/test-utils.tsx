import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { mocked } from 'ts-jest/utils';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext, AuthContextType } from '../context/AuthContext';
import { CFThemeProvider } from 'context';
import { authCtx } from './authCtx';

interface CustomRenderOptions extends RenderOptions {
  initialHistory?: string[];
  authContext?: Partial<AuthContextType>;
}

const customRender = (ui: any, options: CustomRenderOptions = {}) => {
  const { initialHistory, authContext, ...rest } = options;

  // Provide any context that the components may be expecting
  const Wrapper: React.FC = ({ children }) => (
    <CFThemeProvider>
      <MemoryRouter initialEntries={initialHistory}>
        <AuthContext.Provider
          value={{
            ...authCtx,
            ...authContext
          }}
        >
          {children}
        </AuthContext.Provider>
      </MemoryRouter>
    </CFThemeProvider>
  );
  return render(ui, { wrapper: Wrapper, ...rest });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, mocked };
