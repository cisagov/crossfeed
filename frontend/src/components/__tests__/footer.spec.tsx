import React from 'react';
import { Footer } from '../Footer';
import { render } from 'test-utils';

describe('Footer component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
