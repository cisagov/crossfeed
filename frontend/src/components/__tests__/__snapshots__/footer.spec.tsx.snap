// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="makeStyles-root-1"
  >
    Cybersecurity and Infrastructure Security Agency
  </footer>
</DocumentFragment>
`;
